samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManager.scala

```diff
@@ -252,8 +253,30 @@ class KafkaCheckpointManager(
   }
 
   override def start {
+    /*
     kafkaUtil.createTopic(checkpointTopic, 1, replicationFactor, checkpointTopicProperties)
     kafkaUtil.validateTopicPartitionCount(checkpointTopic, systemName, metadataStore, 1, failOnCheckpointValidation)
+    */
+    val CHECKPOINT_STREAMID = "unused-temp-checkpoint-stream-id"
+    val spec = new KafkaStreamSpec(CHECKPOINT_STREAMID,
```

Comment:
`val spec = KafkaStreamSpec.fromSpec(StreamSpec.createCheckpointStreamSpec(checkpointTopic, systemName))`

Then you don't need to introduce another dummy streamId

Reply:
Or you could roll this line into a KafkaStreamSpec static constructor that does the same thing but also takes replication factor:

KafkaStreamSpec.createCheckpointStreamSpec(topicName, systemName, replicationFactor)

That would still allow you to uphold two principles:
1. Immutable StreamSpecs
2. Only one constant for the checkpoint streamId